scene/main/viewport.cpp

```diff
@@ -162,6 +162,24 @@ ViewportTexture::~ViewportTexture() {
 }
 
 void Viewport::_sub_window_update_order() {
+	if (gui.sub_windows.size() < 2)
+		return;
+
+	int index = gui.sub_windows.size() - 1;
+	if (!(gui.sub_windows[index].window->get_flag(Window::FLAG_ALWAYS_ON_TOP))) {
+		SubWindow sw = gui.sub_windows[index];
+
+		for (; index > 0; --index) {
+			if (!(gui.sub_windows[index - 1].window->get_flag(Window::FLAG_ALWAYS_ON_TOP)))
+				break;
+		}
```

Comment:
Not obligatory but I think it would be simpler:
```suggestion
		while (index > 0 && gui.sub_windows[index - 1].window->get_flag(Window::FLAG_ALWAYS_ON_TOP)) {
			--index;
		}
```